commands/cloudapi-v6/datacenter.go

```diff
@@ -240,21 +248,72 @@ func RunDataCenterUpdate(c *core.CommandConfig) error {
 }
 
 func RunDataCenterDelete(c *core.CommandConfig) error {
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete data center"); err != nil {
-		return err
-	}
-	dcId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgDataCenterId))
-	c.Printer.Verbose("Deleting Datacenter with ID: %v...", dcId)
-	resp, err := c.CloudApiV6Services.DataCenters().Delete(dcId)
-	if resp != nil {
-		c.Printer.Verbose(cloudapiv6.RequestTimeMessage, resp.RequestTime)
+	var resp *resources.Response
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv6.ArgAll))
+	if allFlag {
+		err := DeleteAllDatacenters(c)
+		if err != nil {
+			return err
+		}
+	} else {
+		if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete data center"); err != nil {
+			return err
+		}
+		dcId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgDataCenterId))
+		c.Printer.Verbose("Starting deleting Datacenter with ID: %v...", dcId)
+		resp, err := c.CloudApiV6Services.DataCenters().Delete(dcId)
```

Comment:
Use = instead of := to not define resp as a new variable. It will not be recognized at the end of the current function.